set.go

```diff
@@ -82,11 +82,15 @@ func (s *set) Rmv(key string) (*pb.Delta, error) {
 	if err != nil {
 		return nil, err
 	}
-	defer results.Close()
+	defer func() {
+		if results != nil {
+			_ = results.Close()
+		}
+	}()
```

Comment:
I think it was nicer before? Can assume that `results` is not nil at this point. Also, Close is getting called twice.

Reply:
Oops, I left out the bit where I intended to set `results=nil` after calling Close in the happy path.  I agree it was nicer before, except that we wouldn't detect any problems in the event where Close returned an error.  I haven't dug too deeply into whether that's genuine concern or not, which is why I didn't touch the other cases where we ignore errors from results.Close elsewhere.

It looks like the intent of Close is to signal early exit, and it should be safe to ignore the returned error if all results have been processed.  ("Safe" at least insofar as there won't be any lost data, conceivably it could have other implications like resource leaks.)

Reply:
Not that we could do anything about it if there were issues. It's a best-effort "please clean up anything cleanable" type of signal, imho.

Reply:
Got it, thanks.